File: app/src/main/java/com/ads/xinfa/DoubleClickListener.java
package com.ads.xinfa;

import android.view.View;

public abstract class DoubleClickListener implements View.OnClickListener {
    private static final int TIME_DURATION = 1000;
    private static long preTime;

    @Override
    public void onClick(View v) {
        long currentTime = System.currentTimeMillis();
        if (currentTime - preTime < TIME_DURATION) {
            doubleClick(v);
        }
        preTime = currentTime;
    }

    public abstract void doubleClick(View v);

}

File: h5xinfa/src/main/java/com/szty/h5xinfa/baoao/DoubleClickListener.kt
package com.szty.h5xinfa.baoao
import android.view.View

abstract class DoubleClickListener : View.OnClickListener{
    private val TIME_DURATION = 1000
    private var preTime: Long = 0


    override fun onClick(v: View?) {
        val currentTime = System.currentTimeMillis()
        if (currentTime - preTime < TIME_DURATION) {
            doubleClick(v)
        }
        preTime = currentTime
    }

    abstract fun doubleClick(v: View?)
}